compiler/vm.nim

```diff
@@ -144,7 +144,19 @@ proc derefPtrToReg(address: BiggestInt, typ: PType, r: var TFullReg, isAssign: b
   of tyFloat: fun(floatVal, float, rkFloat)
   of tyFloat32: fun(floatVal, float32, rkFloat)
   of tyFloat64: fun(floatVal, float64, rkFloat)
-  else: return false
+  of tyObject:
+    if isAssign:
+      let lhs = cast[PNode](address)
```

Comment:
Wait what?

Reply:
It's the same as all the other pre-existing `cast[ptr T](address)[]` in `derefPtrToReg`, with `T = TNode` in this case.
The tests exercise all of this, and make CT behavior consistent with RT behavior.